controller/aks-cluster-config-handler.go

```diff
@@ -0,0 +1,684 @@
+		if val != "" {
+			tags[key] = to.StringPtr(val)
+		}
+	}
+	displayName := spec.DisplayName
+	if displayName == "" {
+		displayName = spec.ClusterName
+	}
+	tags["displayName"] = to.StringPtr(displayName)
+
+	kubernetesVersion := spec.KubernetesVersion
+	if kubernetesVersion == "" {
+		kubernetesVersion = "1.18.14"
+	}
+
+	//var vmNetSubnetID *string
+	networkProfile := &containerservice.NetworkProfile{}
+	if hasCustomVirtualNetwork(spec) {
+		//virtualNetworkResourceGroup := spec.ResourceGroup
+
+		// if virtual network resource group is set, use it, otherwise assume it is the same as the cluster
+		//if spec.VirtualNetworkResourceGroup != "" {
+		//	virtualNetworkResourceGroup = spec.VirtualNetworkResourceGroup
+		//}
+		/*
+		vmNetSubnetID = to.StringPtr(fmt.Sprintf(
+			"/subscriptions/%v/resourceGroups/%v/providers/Microsoft.Network/virtualNetworks/%v/subnets/%v",
+			spec.SubscriptionID,
+			virtualNetworkResourceGroup,
+			spec.VirtualNetwork,
+			spec.Subnet,
+		))
+		*/
+		networkProfile.DNSServiceIP = to.StringPtr(spec.NetworkDNSServiceIP)
+		networkProfile.DockerBridgeCidr = to.StringPtr(spec.NetworkDockerBridgeCIDR)
+		networkProfile.ServiceCidr = to.StringPtr(spec.NetworkServiceCIDR)
+
+		if spec.NetworkPlugin == "" {
+			networkProfile.NetworkPlugin = containerservice.Azure
+		} else {
+			networkProfile.NetworkPlugin = containerservice.NetworkPlugin(spec.NetworkPlugin)
+		}
+
+		// if network plugin is 'kubenet', set PodCIDR
+		if networkProfile.NetworkPlugin == containerservice.Kubenet {
+			networkProfile.PodCidr = to.StringPtr(spec.NetworkPodCIDR)
+		}
+
+		if spec.NetworkPolicy != "" {
+			networkProfile.NetworkPolicy = containerservice.NetworkPolicy(spec.NetworkPolicy)
+		}
+	}
+
+	loadBalancerSku := containerservice.LoadBalancerSku(spec.LoadBalancerSku)
+	if containerservice.Standard == loadBalancerSku {
+		networkProfile.LoadBalancerSku = loadBalancerSku
+	}
+
+	var agentPoolProfiles []containerservice.ManagedClusterAgentPoolProfile
+	if hasAgentPoolProfile(spec) {
+		for _, ng := range spec.NodeGroups {
+			agentProfile := containerservice.ManagedClusterAgentPoolProfile{
+				Name:         to.StringPtr(ng.Name),
+				Count:        to.Int32Ptr(ng.Count),
+				MaxPods:      to.Int32Ptr(ng.MaxPods),
+				OsDiskSizeGB: to.Int32Ptr(ng.OsDiskSizeGB),
+				OsType:       containerservice.OSType(ng.OsType),
+				VMSize:       containerservice.VMSizeTypes(ng.VMSize),
+				Mode:         containerservice.AgentPoolMode(ng.Mode),
+			}
+			agentPoolProfiles = append(agentPoolProfiles, agentProfile)
+		}
+	} else {
+		agentProfile := containerservice.ManagedClusterAgentPoolProfile{
+			Name:         to.StringPtr("agentpool1"),
+			Count:        to.Int32Ptr(1),
+			MaxPods:      to.Int32Ptr(110),
+			OsDiskSizeGB: to.Int32Ptr(128),
+			OsType:       containerservice.Linux,
+			VMSize:       containerservice.StandardDS2V2,
+			Mode:         containerservice.System,
+		}
+		agentPoolProfiles = append(agentPoolProfiles, agentProfile)
```

Comment:
these kind of default can be displayed in the ui instead of setting them up here and the ng fields on spec should probably be required.

Reply:
Echoing @rmweir 's comment here. For a new cluster there will be a default primary/system node pool but that pool's VM size, node count etc. would have to be based on the user's selection from the Node options section in the rancher ui (in azure portal the system node pool configuration automatically adjusts based on the user's selection in the previous page)

Reply:
This is for creating default test node pool. I would prefer to keep it